scripts/sct_get_centerline.py

```diff
@@ -4,135 +4,157 @@
+        ),
+        formatter_class=SmartFormatter,
+        add_help=None,
+        prog=os.path.basename(__file__).strip(".py")
+    )
+
+    mandatory = parser.add_argument_group("\nMANDATORY ARGUMENTS")
+    mandatory.add_argument(
+        '-i',
+        metavar=Metavar.file,
+        required=True,
+        help="Input image. Example: ti.nii.gz"
+    )
+
+    optional = parser.add_argument_group("\nOPTIONAL ARGUMENTS")
+    optional.add_argument(
+        "-h",
+        "--help",
+        action="help",
+        help="Show this help message and exit."
+    )
+    optional.add_argument(
+        "-c",
+        choices=['t1', 't2', 't2s', 'dwi'],
+        help="Type of image contrast. Only with method=optic."
+    )
+    optional.add_argument(
+        "-method",
+        choices=['optic', 'viewer', 'fitseg'],
+        default='optic',
+        help=("Method used for extracting the centerline.\n"
+              "  - optic: automatic spinal cord detection method\n"
+              "  - viewer: manual selection a few points followed by interpolation\n"
+              "  - fitseg: fit a regularized centerline on an already-existing cord segmentation. It will "
+              "interpolate if slices are missing and extrapolate beyond the segmentation boundaries (i.e., every "
+              "axial slice will exhibit a centerline pixel).")
+    )
+    optional.add_argument(
+        "-centerline-algo",
+        choices=['polyfit', 'bspline', 'linear', 'nurbs'],
+        default='bspline',
+        help="Show this help message and exit."
+    )
+    optional.add_argument(
+        "-centerline-smooth",
+        metavar=Metavar.int,
+        type=int,
+        default=30,
+        help="Degree of smoothing for centerline fitting. Only for -centerline-algo {bspline, linear}."
+    )
+    optional.add_argument(
+        "-o",
+        metavar=Metavar.file,
+        help=("File name (without extension) for the centerline output files. By default, output file will be the "
+              "input with suffix '_centerline'. Example: 'centerline_optic'")
+    )
+    optional.add_argument(
+        "-gap",
+        metavar=Metavar.float,
+        type=float,
+        default=20.0,
+        help="Gap in mm between manually selected points. Only with method=viewer."
+    )
+    optional.add_argument(
+        "-igt",
+        metavar=Metavar.file,
+        help="File name of ground-truth centerline or segmentation (binary nifti)."
+    )
+    optional.add_argument(
+        "-v",
+        choices=['0', '1'],
+        default='1',
+        help="Verbose. 1: display on, 0: display off (default)"
+    )
+    optional.add_argument(
+        "-qc",
+        metavar=Metavar.folder,
+        action=ActionCreateFolder,
+        help="The path where the quality control generated content will be saved."
+    )
+    optional.add_argument(
+        "-qc-dataset",
+        metavar=Metavar.str,
```

Comment:
`Metavar.str` is the default isn't it?

Reply:
If `metavar=Metavar.str` isn't provided, then the example value will instead be:

```
  -qc-dataset QC_DATASET
                        If provided, this string will be mentioned in the QC report as the dataset the
                        process was run on.
```

compared to

```
  -qc-dataset <str>     If provided, this string will be mentioned in the QC report as the dataset the
                        process was run on.
```

I'm happy with either, depending on preference!

Reply:
i like the idea of showing `<str>`, it aligns with the convention for the other flags (and QC_DATASET is not more informative)

Reply:
I find the default behaviour preferable, because it requires less discipline and is more similar to output you would get from other python programs using argparse. I also doubt that most string type arguments are given an explicit metavar across all sct scripts, so this convention would only be partially followed.

Reply:
I don't feel strongly however

Reply:
> I find the default behaviour preferable, because it requires less discipline and is more similar to output you would get from other python programs using argparse. 

This is a fair argument. I was putting myself in the shoes of a typical user of SCT (coming from the neuroimaging community), as opposed to a typical user of other Python software. E.g., I like to often refer to FSL CLI, because SCT users are used to FSL. For example, flirt output shows the type (or the explicit meaning of an input by "paraphrasing" it), example for `flirt`:
~~~
        -in  <inputvol>                    (no default)
        -ref <refvol>                      (no default)
        -init <matrix-filname>             (input 4x4 affine matrix)
        -omat <matrix-filename>            (output in 4x4 ascii format)
        -out, -o <outputvol>               (default is none)
        -datatype {char,short,int,float,double}                    (force output data type)
~~~

> I also doubt that most string type arguments are given an explicit metavar across all sct scripts, so this convention would only be partially followed.

Indeed, most of SCT's conventions are only partially applied because some are fairly recent and we didn't have the chance to fully implement them. Given that the dev team is relatively small, I see the application of convention in a more prospective manner rather than retrospectively (and exhaustively).

Reply:
Ok, I think matching FSL is a reasonable justification